### PR TITLE
fix: refine skipIndex to resolve cyclic dependcy(#29132)

### PR DESCRIPTION
**Pull request by MrPresent-Han**
related: #29132

### CODE REVIEW

File: internal/core/src/index/SkipIndex.cpp [hunk 1]
```diff
@@ -114,10 +123,16 @@ SkipIndex::LoadString(milvus::FieldId field_id,
                 max_string = val;
             }
         }
-        chunkMetrics.min_ = Metrics(min_string);
-        chunkMetrics.max_ = Metrics(max_string);
+        chunkMetrics->min_ = Metrics(min_string);
+        chunkMetrics->max_ = Metrics(max_string);
+    }
```
Comment (MrPresent-Han) [inline internal/core/src/index/SkipIndex.cpp:128]: add mutex_ here to control access towards fieldChunkMetrics

File: internal/core/src/query/visitors/ExecExprVisitor.cpp [hunk 2]
```diff
@@ -36,7 +36,7 @@
 #include "segcore/SegmentGrowingImpl.h"
 #include "simdjson/error.h"
```
Comment (MrPresent-Han) [inline internal/core/src/query/visitors/ExecExprVisitor.cpp:37]: resolve cyclic dependency between milvus_index and milvus_query

**Review by MrPresent-Han — COMMENTED** [review]
comment before reviewing

**Review by yah01 — APPROVED** [review]
/lgtm

**Comment by PowderLi** [comment]
related to #29203

**Pull request merged**